terraform/modules/bosh_vpc/network_private.tf

```diff
@@ -94,6 +94,27 @@ resource "aws_eip" "az2_nat_eip" {
   }
 }
 
+resource "aws_eip" "az1_nat_gateway_eip" {
+  instance = "${aws_instance.az1_private_nat_2017_09.id}"
+  vpc = true
+
+  count = "${var.use_nat_gateway_eip == "true" ? 1 : 0}"
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+resource "aws_eip" "az2_nat_gateway_eip" {
+  instance = "${aws_instance.az2_private_nat_2017_09.id}"
+  vpc = true
+
+  count = "${var.use_nat_gateway_eip == "true" ? 1 : 0}"
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
```

Comment:
hmmm, not sure about this at all, since these EIPs are tied to instances and `aws_nat_gateway` is a whole 'nother thing.